addons/hr_expense/models/hr_expense.py

```diff
@@ -349,6 +349,8 @@ def message_new(self, msg_dict, custom_values=None):
         else:
             expense_description = expense_description.replace(product_code.group(), '')
             product = self.env['product.product'].search([('default_code', 'ilike', product_code.group(1))]) or default_product
+            if len(product) > 1:
+                product = product.filtered(lambda p: p.default_code == product_code.group(1)) or product[0]
```

Comment:
the `if` is not really needed, worst-case scenario you run the filtered on a single record.

ps: if you need a [mailcatcher](https://mailcatcher.me/)

Reply:
Thanks you are right
Was to make the purpose of the line more clear but renaming the first `product` variable to `products` do the trick too